pydra/design/base.py

```diff
@@ -244,12 +244,14 @@ class Arg(Field):
         the default value for the field, by default it is NO_DEFAULT
     help: str
         A short description of the input field.
-    allowed_values: list, optional
-        List of allowed values for the field.
     requires: list, optional
         Names of the inputs that are required together with the field.
-    xor: list[str], optional
-        Names of the inputs that are mutually exclusive with the field.
+    allowed_values: Sequence, optional
+        List of allowed values for the field.
+    xor: Sequence[str | None], optional
+        Names of args that are exclusive mutually exclusive, which must include
+        the name of the current field. If this list includes None, then none of the
```

Comment:
including the current field was not the case in nipype or the previous version of pydra i believe. reducing this kind of boilerplate would be helpful, and also if `xor` is commutative without specification (`a` xor `b` implies `b` xor `a` and should not have to be specified in both places - just one should do the other should be computed). also any checks to make sure xor combinations are valid or invalid (e.g. `a` xor `b`, `b` xor `c`, `c` xor `a`)

Reply:
I was thinking it would be better as an argument to the `shell.define` decorator/function, so it would only need to be defined once and is really a "task-level" attribute

Reply:
so a list of lists? i.e. there could be multiple different `xor`s in a task.

Reply:
Yes, maybe list[str] | list[list[str]] could be acceptable inputs